### ISSUE TITLE
regress arc4randomforktest fails on Cygwin

### ISSUE THREAD

**Issue by kinichiro** [libressl-portable/portable#289]
I found arc4randomforktest fails on Cygwin. Here is the log message.

```
$ cat arc4randomforktest.sh.log
assertion "(countone) <= (1)" failed: file "arc4randomforktest.c", line 166, function: main
./arc4randomforktest.sh: 5 :  7448 Aborted                 (coredump) ./arc4randomforktest -p
```

It appears that flag "-p" causes failure.
I confirmed this issue with libressl current master branch with Cygwin64 2.6.0.

```
$ uname -a
CYGWIN_NT-6.1 XXXXXXX 2.6.0(0.304/5/3) 2016-08-31 14:32 x86_64 Cygwin
```

Before, I thought this regression test had been succeeding on Cygwin, though.

**Comment by kinichiro** [comment]
This issue appears still remain with Cygwin64 2.7.0.

```
$ uname -a
CYGWIN_NT-6.1 testpc001 2.7.0(0.306/5/3) 2017-02-12 13:18 x86_64 Cygwin
```

**Comment by kinichiro** [comment]
This issue still remain with 2.8.0.

```
$ uname -a
CYGWIN_NT-6.1 testpc001 2.8.0(0.309/5/3) 2017-04-01 20:47 x86_64 Cygwin
```

I posted this to Cygwin ML.
https://cygwin.com/ml/cygwin/2017-04/msg00368.html

**Comment by busterb** [comment]
That's a pretty lame response from the Cygwin team.

**Comment by busterb** [comment]
This looks like an actual bug in the cygwin implementation of arc4random in detecting fork wraparound. Though I'd say that's a pretty minor corner case, even though Ars Technica blew it up 3 years ago (https://arstechnica.com/information-technology/2014/07/only-a-few-days-old-openssl-fork-libressl-is-declared-unsafe-for-linux/). It is still not a bug necessarily in LibreSSL, and I'd definitely prefer that it was fixed upstream so other software can also take advantage of the fixes. See https://www.agwa.name/blog/post/libressls_prng_is_unsafe_on_linux. (Note that we fixed this July 2014).

I would suggest following up to the Cygwin mailing list thread with a link to this info. Cygwin's arc4random implementation is vulnerable to the same issue, and the test code is usable outside of LibreSSL as well.
